unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/AvatarEmotesController.cs

```diff
@@ -83,16 +86,27 @@ private async UniTask AsyncEmoteLoad(string bodyShapeId, string emoteId)
             catch (Exception e) { Debug.LogException(e); }
         }
 
-        public void PlayEmote(string emoteId, long timestamps, bool spatial, bool occlude, bool ignoreTimestamp)
+        public void PlayEmote(string emoteId, long timestamps, bool spatial, bool occlude, bool forcePlay)
         {
             if (string.IsNullOrEmpty(emoteId)) return;
-            if (visibilityConstraints.Count > 0) return;
+            if (!CanPlayEmote()) return;
 
             var emoteKey = new EmoteBodyId(bodyShapeId, emoteId);
             if (!equippedEmotes.ContainsKey(emoteKey)) return;
 
             float volume = GetEmoteVolume();
-            animator.PlayEmote(emoteId, timestamps, spatial, volume, occlude, ignoreTimestamp);
+            animator.PlayEmote(emoteId, timestamps, spatial, volume, occlude, forcePlay);
+        }
+
+        private bool CanPlayEmote()
+        {
+            var isHiddenByConstraint = false;
+
+            foreach (string constraint in visibilityConstraints)
+                if (constraint != INSIDE_CAMERA)
+                    isHiddenByConstraint = true;
+
+            return !isHiddenByConstraint;
```

Comment:
Small suggestion, we could simplify this and make it more readable, as `visibilityConstraints` is Hashset there will be no impact on performance over iteration with Contains.
```suggestion
            if (visibilityConstraints.Count == 0) return true;
            if (visibilityConstraints.Count > 1) return false;
            return visibilityConstraints.Contains(INSIDE_CAMERA);
```